docs/content/dev/design/multiple-url-to-same-repo.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Matching multiple URLs to the same Repository CR
+weight: 1
+---
+
+# Matching multiple URLs to the same Repository CR
+
+## Background
+
+The current implementation of the Repository CR allows specifying a single URL
+for the repository. This is limiting in cases where the repository is mirrored
+to multiple URLs. This document proposes a way to allow specifying multiple URLs
```

Comment:
creating one CR for each repo?
is this a lot of work for this use case? or sharing the settings? across repos?

Reply:
I listed this in the alternatives sections that we could potentially reuse the configmap features of that setting you implemented sometime ago and extend it, but that's an admin thing that doesn't allow much flexibility...

sharing setting is an interesting idea,having a configmap reference that would embed the value..

Reply:
I think allowing multiple repo url could be a nice thing but if we allow regex then it might become quite complex as you have defined in scenarios
handling cases with multiple repos, what if a repo is edited and all

if the requirement for this design is to somehow share settings across repos then we could find someother way 🤔

Reply:
I don't think there is flow in the scenarios i have listed, it's just a matter to have the webhook check if the new value (on update or creation) matches already and fail otherwise? is there more complexities to it do you think?